pkgs/build-support/dart/pub2nix/package-graph/generator.py

```diff
@@ -0,0 +1,47 @@
+import json
+import os
+from pathlib import Path
+
+import yaml
+
+
+def main() -> None:
+    attrs_file_path = Path(os.environ["NIX_ATTRS_JSON_FILE"])
+    with attrs_file_path.open("r", encoding="utf-8") as f:
+        attrs = json.load(f)
+    package_graphs = []
+    for name, data in attrs.get("packages", {}).items():
+        with (
+            Path(data["src"])
+            / data.get("packageRoot", ".").lstrip("/")
+            / "pubspec.yaml"
+        ).open("r", encoding="utf-8") as f:
+            pubspec = yaml.safe_load(f)
+            package_graphs.append({
+                "name": name,
+                "version": pubspec.get("version") or "0.0.0",
+                "dependencies": list((pubspec.get("dependencies") or {}).keys()),
+            })
+    with Path(attrs.get("pubspecFile")).open("r", encoding="utf-8") as f:
+        pubspec = yaml.safe_load(f)
+        package_graphs.append({
+            "name": pubspec.get("name"),
+            "version": pubspec.get("version") or "0.0.0",
+            "dependencies": list((pubspec.get("dependencies") or {}).keys()),
+            "devDependencies": list((pubspec.get("dev_dependencies") or {}).keys()),
+        })
```

Comment:
Nit - there's a little bit of code here that may be clearer deduplicated.

The logic to parse a `pubspec.yaml` and generate an object for `package_graphs` can be moved into a function with a conditional parameter to include `devDependencies`.

This would more obviously separate the actions of generating the list of `pubspec.yaml`s and generating the package graph contents.